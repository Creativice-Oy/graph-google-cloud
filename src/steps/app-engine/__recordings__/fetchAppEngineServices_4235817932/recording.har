{
  "log": {
    "_recordingName": "fetchAppEngineServices",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "acea721c8193b51ced888cae721cc423",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 739,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "content-length",
              "value": "739"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "www.googleapis.com"
            }
          ],
          "headersSize": 283,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "postData": {
            "mimeType": "application/json",
            "params": [],
            "text": "[REDACTED]"
          },
          "queryString": [],
          "url": "https://www.googleapis.com/oauth2/v4/token"
        },
        "response": {
          "bodySize": 565,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 565,
            "text": "{\"access_token\":\"[REDACTED]\",\"expires_in\":9999,\"token_type\":\"Bearer\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Thu, 11 Mar 2021 14:57:50 GMT"
            },
            {
              "name": "server",
              "value": "scaffolding on HTTPServer2"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 511,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-03-11T14:57:50.653Z",
        "time": 170,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 170
        }
      },
      {
        "_id": "a8077ca10a73994713b7a0d586480166",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/4.4.3 gl-node/14.15.4 auth/6.0.5"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/4.4.3 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "appengine.googleapis.com"
            }
          ],
          "headersSize": 538,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://appengine.googleapis.com/v1/apps/j1-gc-integration-dev-v2"
        },
        "response": {
          "bodySize": 637,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 637,
            "text": "{\"name\":\"apps/j1-gc-integration-dev-v2\",\"id\":\"j1-gc-integration-dev-v2\",\"authDomain\":\"gmail.com\",\"locationId\":\"us-central\",\"codeBucket\":\"staging.j1-gc-integration-dev-v2.appspot.com\",\"servingStatus\":\"SERVING\",\"defaultHostname\":\"j1-gc-integration-dev-v2.uc.r.appspot.com\",\"defaultBucket\":\"j1-gc-integration-dev-v2.appspot.com\",\"gcrDomain\":\"us.gcr.io\",\"databaseType\":\"CLOUD_DATASTORE_COMPATIBILITY\",\"featureSettings\":{\"splitHealthChecks\":true,\"useContainerOptimizedOs\":true}}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Thu, 11 Mar 2021 14:57:51 GMT"
            },
            {
              "name": "server",
              "value": "ESF"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 488,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-03-11T14:57:50.835Z",
        "time": 455,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 455
        }
      },
      {
        "_id": "acea721c8193b51ced888cae721cc423",
        "_order": 1,
        "cache": {},
        "request": {
          "bodySize": 739,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "content-length",
              "value": "739"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "www.googleapis.com"
            }
          ],
          "headersSize": 283,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "postData": {
            "mimeType": "application/json",
            "params": [],
            "text": "[REDACTED]"
          },
          "queryString": [],
          "url": "https://www.googleapis.com/oauth2/v4/token"
        },
        "response": {
          "bodySize": 552,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 552,
            "text": "{\"access_token\":\"[REDACTED]\",\"expires_in\":9999,\"token_type\":\"Bearer\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Thu, 11 Mar 2021 14:58:50 GMT"
            },
            {
              "name": "server",
              "value": "scaffolding on HTTPServer2"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 511,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-03-11T14:58:49.917Z",
        "time": 169,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 169
        }
      },
      {
        "_id": "83742efc2d14453e4066991a3f93e347",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/4.4.3 gl-node/14.15.4 auth/6.0.5"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/4.4.3 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "appengine.googleapis.com"
            }
          ],
          "headersSize": 547,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://appengine.googleapis.com/v1/apps/j1-gc-integration-dev-v2/services"
        },
        "response": {
          "bodySize": 311,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 311,
            "text": "{\"services\":[{\"name\":\"apps/j1-gc-integration-dev-v2/services/default\",\"id\":\"default\",\"split\":{\"allocations\":{\"20210309t141022\":1}}}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Thu, 11 Mar 2021 14:58:50 GMT"
            },
            {
              "name": "server",
              "value": "ESF"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 488,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-03-11T14:58:50.091Z",
        "time": 461,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 461
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
