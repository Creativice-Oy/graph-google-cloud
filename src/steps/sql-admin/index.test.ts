import { createMockStepExecutionContext } from '@jupiterone/integration-sdk-testing';
import { fetchCloudSQLInstances } from '.';
import { integrationConfig } from '../../../test/config';
import { Recording, setupGoogleCloudRecording } from '../../../test/recording';
import { IntegrationConfig } from '../../types';
import { fetchResourceManagerProject } from '../resource-manager';

import {
  CLOUD_SQL_ADMIN_MYSQL_INSTANCE_ENTITY_TYPE,
  CLOUD_SQL_ADMIN_POSTGRES_INSTANCE_ENTITY_TYPE,
  CLOUD_SQL_ADMIN_SQL_SERVER_INSTANCE_ENTITY_TYPE,
} from './constants';

describe('#fetchSQLInstances', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupGoogleCloudRecording({
      directory: __dirname,
      name: 'fetchSQLInstances',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {

    const customConfig = { ...integrationConfig, serviceAccountKeyConfig: { ...integrationConfig.serviceAccountKeyConfig, project_id: 'j1-gc-integration-dev-300716' }}
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: customConfig,
    });

    await fetchResourceManagerProject(context);
    await fetchCloudSQLInstances(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      numCollectedRelationships: context.jobState.collectedRelationships.length,
      collectedEntities: context.jobState.collectedEntities,
      collectedRelationships: context.jobState.collectedRelationships,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    expect(
      context.jobState.collectedEntities.filter(
        (e) => e._type === CLOUD_SQL_ADMIN_MYSQL_INSTANCE_ENTITY_TYPE,
      ),
    ).toMatchGraphObjectSchema({
      _class: ['Database'],
      schema: {
        additionalProperties: false,
        properties: {
          _type: { const: 'google_cloud_sql_mysql_instance' },
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          name: { type: 'string' },
          type: { type: 'string' },
          location: { type: 'string' },
          displayName: { type: 'string' },
          localInfile: { type: 'string' },
          requireSSL: { type: 'boolean' },
          authorizedNetworks: {
            type: 'array',
            items: { type: 'string' },
          },
          hasPublicIP: { type: 'boolean' },
          automatedBackupsEnabled: { type: 'boolean' },
        },
      },
    });

    expect(
      context.jobState.collectedEntities.filter(
        (e) => e._type === CLOUD_SQL_ADMIN_POSTGRES_INSTANCE_ENTITY_TYPE,
      ),
    ).toMatchGraphObjectSchema({
      _class: ['Database'],
      schema: {
        additionalProperties: false,
        properties: {
          _type: { const: 'google_cloud_sql_postgres_instance' },
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          name: { type: 'string' },
          type: { type: 'string' },
          location: { type: 'string' },
          displayName: { type: 'string' },
          logCheckpoints: { type: 'string' },
          logConnections: { type: 'string' },
          logDisconnections: { type: 'string' },
          logLockWaits: { type: 'string' },
          logMinErrorStatement: { type: 'string' },
          logTempFiles: { type: 'string' },
          logMinDurationStatement: { type: 'string' },
          requireSSL: { type: 'boolean' },
          authorizedNetworks: {
            type: 'array',
            items: { type: 'string' },
          },
          hasPublicIP: { type: 'boolean' },
          automatedBackupsEnabled: { type: 'boolean' },
        },
      },
    });

    expect(
      context.jobState.collectedEntities.filter(
        (e) => e._type === CLOUD_SQL_ADMIN_SQL_SERVER_INSTANCE_ENTITY_TYPE,
      ),
    ).toMatchGraphObjectSchema({
      _class: ['Database'],
      schema: {
        additionalProperties: false,
        properties: {
          _type: { const: 'google_cloud_sql_sql_server_instance' },
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          name: { type: 'string' },
          type: { type: 'string' },
          location: { type: 'string' },
          displayName: { type: 'string' },
          crossDatabaseOwnershipChaining: { type: 'string' },
          containedDatabaseAuthentication: { type: 'string' },
          requireSSL: { type: 'boolean' },
          authorizedNetworks: {
            type: 'array',
            items: { type: 'string' },
          },
          hasPublicIP: { type: 'boolean' },
          automatedBackupsEnabled: { type: 'boolean' },
        },
      },
    });
  });
});
