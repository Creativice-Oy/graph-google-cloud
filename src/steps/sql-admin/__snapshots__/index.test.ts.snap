// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#fetchSQLInstances should collect data 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "Account",
      ],
      "_key": "j1-gc-integration-dev-300716",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "createTime": "2021-01-04T16:56:24.926Z",
            "lifecycleState": "ACTIVE",
            "name": "j1-gc-integration-dev",
            "parent": Object {
              "id": "590372371675",
              "type": "organization",
            },
            "projectId": "j1-gc-integration-dev-300716",
            "projectNumber": "165882964161",
          },
        },
      ],
      "_type": "google_cloud_project",
      "createdOn": 1609779384926,
      "displayName": "j1-gc-integration-dev",
      "lifecycleState": "ACTIVE",
      "name": "j1-gc-integration-dev",
      "parent.id": "590372371675",
      "parent.type": "organization",
      "projectNumber": "165882964161",
    },
    Object {
      "_class": Array [
        "Database",
      ],
      "_key": "j1-gc-integration-dev-300716:us-central1:cloud-sql-postgres-b150e56d8488b376",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "backendType": "SECOND_GEN",
            "connectionName": "j1-gc-integration-dev-300716:us-central1:cloud-sql-postgres-b150e56d8488b376",
            "databaseVersion": "POSTGRES_12",
            "etag": "fec1e7299714066c6fa530a2d645741bbc535d38bd0eab7074051d4455fd1a4e",
            "gceZone": "us-central1-f",
            "instanceType": "CLOUD_SQL_INSTANCE",
            "ipAddresses": Array [
              Object {
                "ipAddress": "35.238.236.140",
                "type": "PRIMARY",
              },
            ],
            "kind": "sql#instance",
            "name": "cloud-sql-postgres-b150e56d8488b376",
            "project": "j1-gc-integration-dev-300716",
            "region": "us-central1",
            "selfLink": "https://www.googleapis.com/sql/v1beta4/projects/j1-gc-integration-dev-300716/instances/cloud-sql-postgres-b150e56d8488b376",
            "serverCaCert": Object {
              "cert": "-----BEGIN CERTIFICATE-----
MIIDfzCCAmegAwIBAgIBADANBgkqhkiG9w0BAQsFADB3MS0wKwYDVQQuEyQxYjIy
ZmE0YS1iYWE4LTQ0NmUtYjdhMC0yMzM3Yzg2ZTE1YmExIzAhBgNVBAMTGkdvb2ds
ZSBDbG91ZCBTUUwgU2VydmVyIENBMRQwEgYDVQQKEwtHb29nbGUsIEluYzELMAkG
A1UEBhMCVVMwHhcNMjEwMTE3MjIyMTE0WhcNMzEwMTE1MjIyMjE0WjB3MS0wKwYD
VQQuEyQxYjIyZmE0YS1iYWE4LTQ0NmUtYjdhMC0yMzM3Yzg2ZTE1YmExIzAhBgNV
BAMTGkdvb2dsZSBDbG91ZCBTUUwgU2VydmVyIENBMRQwEgYDVQQKEwtHb29nbGUs
IEluYzELMAkGA1UEBhMCVVMwggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIB
AQCZd6VzhW5aLVuoZ1zc4RWT02Gg+6Px1fxsOYqzccZ5rYLr/BdjP1RoPAtLQwEA
2dflZn9N1KAEraCorppXTa0od3xpFQrMIYi2zLuyXZ3s8pQ3EHdVPQoa2PVkbGOH
bVozRlPGhnwO/O5/zbEaZasrOeO4m0qOzIVEcwO4MGYOtiI49o3qKbWoF/wf7JsG
zWomAVvQQdkhBprTy+E2JgJxN8P/clQ/0aJQTs3FZMwvEL+BElzMQJPGiH02SqRT
KiPx1Q++a9xGeV2BVzwU7BEIxCkRmLYd2ydCASvbt0lLB6zYCIqzy0nvf+NlPkpq
fTt1h+Fp5AQMz9dvsD/jKzOBAgMBAAGjFjAUMBIGA1UdEwEB/wQIMAYBAf8CAQAw
DQYJKoZIhvcNAQELBQADggEBABxTljxkWOXoKCc57cCsrmHtg2gK3DLkalJuyqnY
/8p6bSHTqK31v6FysgzHLrCU5efOVW7NsRUBafuNAU6244b05RtjOF6a5xPJLVfx
cyY0krciDi0IE5gOgCr/f+0R3Y19y/+PqYnogkMXshdOD7h9VwqyLCaGsuc+Mjcx
COvJ6FC+TuGDBJz+KnxMPIZgrB3b1mpAjRFrvW0yJYi8iRwvxvShmhz6QxpEXY/0
xt8kXLb2r/M5krTTh/mp28/0tgc85mJPbVLo6blLyHrwxBBnpWHKCBIr98YReu+I
91KU/+mokZnf2hQWVMyUP+Fnt/9cuOPgDV3dgMjFwgO/pGM=
-----END CERTIFICATE-----",
              "certSerialNumber": "0",
              "commonName": "C=US,O=Google\\\\, Inc,CN=Google Cloud SQL Server CA,dnQualifier=1b22fa4a-baa8-446e-b7a0-2337c86e15ba",
              "createTime": "2021-01-17T22:21:14.708Z",
              "expirationTime": "2031-01-15T22:22:14.708Z",
              "instance": "cloud-sql-postgres-b150e56d8488b376",
              "kind": "sql#sslCert",
              "sha1Fingerprint": "fdf1f3e2e657dd41b0f184f40f31135de1783cf9",
            },
            "serviceAccountEmailAddress": "p165882964161-zydq2b@gcp-sa-cloud-sql.iam.gserviceaccount.com",
            "settings": Object {
              "activationPolicy": "ALWAYS",
              "authorizedGaeApplications": Array [],
              "availabilityType": "ZONAL",
              "backupConfiguration": Object {
                "backupRetentionSettings": Object {
                  "retainedBackups": 7,
                  "retentionUnit": "COUNT",
                },
                "enabled": false,
                "kind": "sql#backupConfiguration",
                "location": "us",
                "pointInTimeRecoveryEnabled": false,
                "replicationLogArchivingEnabled": false,
                "startTime": "14:00",
                "transactionLogRetentionDays": 7,
              },
              "dataDiskSizeGb": "10",
              "dataDiskType": "PD_SSD",
              "databaseFlags": Array [
                Object {
                  "name": "log_connections",
                  "value": "on",
                },
              ],
              "ipConfiguration": Object {
                "authorizedNetworks": Array [],
                "ipv4Enabled": true,
              },
              "kind": "sql#settings",
              "locationPreference": Object {
                "kind": "sql#locationPreference",
                "zone": "us-central1-f",
              },
              "maintenanceWindow": Object {
                "day": 0,
                "hour": 0,
                "kind": "sql#maintenanceWindow",
              },
              "pricingPlan": "PER_USE",
              "replicationType": "SYNCHRONOUS",
              "settingsVersion": "2",
              "storageAutoResize": true,
              "storageAutoResizeLimit": "0",
              "tier": "db-f1-micro",
            },
            "state": "RUNNABLE",
          },
        },
      ],
      "_type": "google_sql_postgres_instance",
      "authorizedNetworks": Array [],
      "automatedBackupsEnabled": false,
      "createdOn": undefined,
      "displayName": "cloud-sql-postgres-b150e56d8488b376",
      "encrypted": true,
      "hasPublicIP": true,
      "location": "j1-gc-integration-dev-300716:us-central1:cloud-sql-postgres-b150e56d8488b376",
      "logCheckpoints": undefined,
      "logConnections": "on",
      "logDisconnections": undefined,
      "logLockWaits": undefined,
      "logMinDurationStatement": undefined,
      "logMinErrorStatement": undefined,
      "logTempFiles": undefined,
      "name": "cloud-sql-postgres-b150e56d8488b376",
      "requireSSL": undefined,
    },
    Object {
      "_class": Array [
        "Database",
      ],
      "_key": "j1-gc-integration-dev-300716:us-central1:cloud-sql-mysql-b150e56d8488b376",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "backendType": "SECOND_GEN",
            "connectionName": "j1-gc-integration-dev-300716:us-central1:cloud-sql-mysql-b150e56d8488b376",
            "databaseVersion": "MYSQL_5_7",
            "etag": "5f0d52ebc3759f6d21b11e1253de9950f0da3f1f482f6384a61ffbd87683df65",
            "gceZone": "us-central1-f",
            "instanceType": "CLOUD_SQL_INSTANCE",
            "ipAddresses": Array [
              Object {
                "ipAddress": "104.197.53.88",
                "type": "PRIMARY",
              },
            ],
            "kind": "sql#instance",
            "name": "cloud-sql-mysql-b150e56d8488b376",
            "project": "j1-gc-integration-dev-300716",
            "region": "us-central1",
            "selfLink": "https://www.googleapis.com/sql/v1beta4/projects/j1-gc-integration-dev-300716/instances/cloud-sql-mysql-b150e56d8488b376",
            "serverCaCert": Object {
              "cert": "-----BEGIN CERTIFICATE-----
MIIDfzCCAmegAwIBAgIBADANBgkqhkiG9w0BAQsFADB3MS0wKwYDVQQuEyQ1NWVi
MzRjNy1hZTYxLTQ3NmEtYjQ4Yi1hNjFmN2Y3OGNhYmYxIzAhBgNVBAMTGkdvb2ds
ZSBDbG91ZCBTUUwgU2VydmVyIENBMRQwEgYDVQQKEwtHb29nbGUsIEluYzELMAkG
A1UEBhMCVVMwHhcNMjEwMTE3MjMwOTU5WhcNMzEwMTE1MjMxMDU5WjB3MS0wKwYD
VQQuEyQ1NWViMzRjNy1hZTYxLTQ3NmEtYjQ4Yi1hNjFmN2Y3OGNhYmYxIzAhBgNV
BAMTGkdvb2dsZSBDbG91ZCBTUUwgU2VydmVyIENBMRQwEgYDVQQKEwtHb29nbGUs
IEluYzELMAkGA1UEBhMCVVMwggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIB
AQC1i6x41zra53w9lMklwlrPBE2qpNUnyzuB5Qj1DlJIZwh7sFZpwXsbNTL/U/Kd
YxTCLBAIZqmKx16+takurHuBSemGMROWiP7KBbUH+Mh6Xlg0gVaKtD00eG9zBUAu
SYy8S5OVu/sylQ5ZyxFj5ZLXknarqcqhvIzmwdk6pSR6pGYoZRQ3F04SembrJaL2
RdufAZ2kQ6Yi9HQ5nyTkyG5tBomOQK/GX2iWvxl7UkW1NZuyOf4StRgK6GrvJ5qr
wvZoZ068QA3SywDl/ZCwSHobTNqezIj2qG8ZSjDAmqkPnjpaIUl5YRW0Mi/A4q+W
Z1nKv+rTStpzHz0ySIy/aEePAgMBAAGjFjAUMBIGA1UdEwEB/wQIMAYBAf8CAQAw
DQYJKoZIhvcNAQELBQADggEBAIb48C8tY6JbALYRNoJCkjzAujEWAGv9qN9f47IN
ydN+NFbd5tYuxrDrMI0hMz+xQMswTBYmtwzLfQjDtmm99Ho0axktAEDqtuoHbQnF
FY2deikKvCYZ3nInnJOctFY3mIhYWLKJxlWNjxv4up9PhTsEJehWlrcp8VJh0JQw
Lwyd9MLA6stoLHUEaLpuFa0u75y8lmnaiN6GvdukbGUMtG5gEBa4biUCBhBqaQeK
0eK25+lZD3n/dnoCEZlY33Meiz225uoCqPnk6jDPLKlm3T/fj1kFpna9gyOMTa/p
1VxeU/xJre0p893x6ggoPJ0yCWH1Tx+5nP7JOuqxNaoZHdA=
-----END CERTIFICATE-----",
              "certSerialNumber": "0",
              "commonName": "C=US,O=Google\\\\, Inc,CN=Google Cloud SQL Server CA,dnQualifier=55eb34c7-ae61-476a-b48b-a61f7f78cabf",
              "createTime": "2021-01-17T23:09:59.121Z",
              "expirationTime": "2031-01-15T23:10:59.121Z",
              "instance": "cloud-sql-mysql-b150e56d8488b376",
              "kind": "sql#sslCert",
              "sha1Fingerprint": "1c070a5b398b53a6d871534acf8dbba58d732ab0",
            },
            "serviceAccountEmailAddress": "p165882964161-gqysvy@gcp-sa-cloud-sql.iam.gserviceaccount.com",
            "settings": Object {
              "activationPolicy": "ALWAYS",
              "authorizedGaeApplications": Array [],
              "availabilityType": "ZONAL",
              "backupConfiguration": Object {
                "backupRetentionSettings": Object {
                  "retainedBackups": 7,
                  "retentionUnit": "COUNT",
                },
                "enabled": false,
                "kind": "sql#backupConfiguration",
                "startTime": "02:00",
                "transactionLogRetentionDays": 7,
              },
              "dataDiskSizeGb": "10",
              "dataDiskType": "PD_SSD",
              "ipConfiguration": Object {
                "authorizedNetworks": Array [
                  Object {
                    "kind": "sql#aclEntry",
                    "name": "0.0.0.0/0",
                    "value": "0.0.0.0/0",
                  },
                ],
                "ipv4Enabled": true,
              },
              "kind": "sql#settings",
              "locationPreference": Object {
                "kind": "sql#locationPreference",
                "zone": "us-central1-f",
              },
              "pricingPlan": "PER_USE",
              "replicationType": "SYNCHRONOUS",
              "settingsVersion": "3",
              "storageAutoResize": true,
              "storageAutoResizeLimit": "0",
              "tier": "db-f1-micro",
            },
            "state": "RUNNABLE",
          },
        },
      ],
      "_type": "google_sql_mysql_instance",
      "authorizedNetworks": Array [
        "0.0.0.0/0",
      ],
      "automatedBackupsEnabled": false,
      "createdOn": undefined,
      "displayName": "cloud-sql-mysql-b150e56d8488b376",
      "encrypted": true,
      "hasPublicIP": true,
      "localInfile": undefined,
      "location": "j1-gc-integration-dev-300716:us-central1:cloud-sql-mysql-b150e56d8488b376",
      "name": "cloud-sql-mysql-b150e56d8488b376",
      "requireSSL": undefined,
    },
  ],
  "collectedRelationships": Array [],
  "encounteredTypes": Array [
    "google_cloud_project",
    "google_sql_postgres_instance",
    "google_sql_mysql_instance",
  ],
  "numCollectedEntities": 3,
  "numCollectedRelationships": 0,
}
`;
